index.js

```diff
@@ -3,89 +3,115 @@
 // # S3 storage module for Ghost blog http://ghost.org/
 var fs = require('fs');
 var path = require('path');
-var nodefn = require('when/node/function');
-var when = require('when');
-var readFile = nodefn.lift(fs.readFile);
-var unlink = nodefn.lift(fs.unlink);
-var AWS = require('aws-sdk');
+var Bluebird = require('bluebird');
```

Comment:
Ghost uses [Bluebird](http://bluebirdjs.com/), which provides some very useful promise creation mechanisms. I changed the code to use these